src/gtest/test_utils.cpp

```diff
@@ -0,0 +1,67 @@
+#include <gtest/gtest.h>
+
+#include "tx_creation_utils.h"
+#include "zen/forks/fork9_sidechainversionfork.h"
+
+using namespace blockchain_test_utils;
+
+class BlockchainHelperTest: public ::testing::Test
+{
+protected:
+
+    int sidechainForkHeight;
+
+    BlockchainHelperTest()
+    {
+        SidechainFork sidechainFork = SidechainFork();
+        sidechainForkHeight = sidechainFork.getHeight(CBaseChainParams::REGTEST);
+    }
+
+    void SetUp() override
+    {
+        SelectParams(CBaseChainParams::REGTEST);
+    }
+
+    void TearDown() override
+    {
+        chainActive.SetTip(nullptr);
+    }
+};
+
+/**
+ * This test is intended to check that the BlockchainTestManager class behaves as expected
+ * also after calling the "Reset()" function.
+ * In particular, it checks that the transaction creation works when requesting to generate
+ * input coins.
+ */
+TEST_F(BlockchainHelperTest, CoinGeneration)
+{
+    BlockchainTestManager& testManager = BlockchainTestManager::GetInstance();
+
+    // Initialize the sidechain keys
+    testManager.GenerateSidechainTestParameters(ProvingSystem::CoboundaryMarlin, TestCircuitType::Certificate);
+
+    // Go to the last block before the SidechainVersionFork
+    
+    testManager.ExtendChainActiveToHeight(sidechainForkHeight);
```

Comment:
I'd suggest to add a comment to explain if and why a specific height is necessary for this test.